### PR TITLE
softmaxold layer

### PR DESCRIPTION
**Pull request by nopphons**
Added softmax old layer from old version of caffe

### CODE REVIEW

File: include/caffe/common_layers.hpp [hunk 1]
```diff
@@ -108,6 +109,40 @@ class BatchNormLayer : public Layer<Dtype> {
 
 };
 
+template <typename Dtype>
+class FixedBatchNormLayer : public Layer<Dtype> {
```
Comment (cbfinn) [inline include/caffe/common_layers.hpp]: What is this?

File: include/caffe/common_layers.hpp [hunk 2]
```diff
@@ -519,6 +554,65 @@ class CuDNNSoftmaxLayer : public SoftmaxLayer<Dtype> {
...
+  string dimension_;
+};
+
+#ifdef USE_CUDNN
+/**
+ * @brief cuDNN implementation of SoftmaxLayer.
+ *        Fallback to SoftmaxLayer for CPU mode.
+ */
+template <typename Dtype>
+class CuDNNSoftmaxOldLayer : public SoftmaxOldLayer<Dtype> {
```
Comment (cbfinn) [inline include/caffe/common_layers.hpp]: The old CuDNN version isn't needed. It doesn't include a spatial softmax.

**Comment by cbfinn** [comment]
Can you not include the old CuDNN softmax layer? (Is it needed for the normal softmax layer?) Also, the fixed batch norm layer belongs in a different pull request if you want to merge it.

Otherwise, lgtm. (Feel free to merge once you've deleted the old cudnn layer and fixed batch norm layer from this PR.)

**Comment by nopphons** [comment]
Yes, the fixed batch norm was a mistake.
I think it is needed when the engine is not CAFFE, but I can remove it.
